code/processes/wdb.q

```diff
@@ -296,8 +308,10 @@ merge:{[dir;pt;tableinfo;mergelimits]
 endofdaymerge:{[dir;pt;tablist;mergelimits]		
 	/- merge data from partitons
 	$[(0 < count .z.pd[]) and ((system "s")<0);
-		[.lg.o[`merge;"merging on slave"];
+		[.lg.o[`merge;"merging on slave"];		
+		setcompression[compression];
 		merge[dir;pt;;mergelimits] peach flip (key tablist;value tablist);];	
+		removecompression[compression];
```

Comment:
I don't think this will work. setcompression will be executed locally, whereas merge will be executed remotely.  You need to wrap the merge statement in a lambda and set/remove compression either side of the merge function call (the same as you do with the other peach statement)